### PR TITLE
Adding owner column for secrets

### PR DESCRIPTION
**Pull request by mmontgomery-square**
…l integrity

### CODE REVIEW

File: server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE secrets ADD owner BIGINT;
+ALTER TABLE secrets ADD CONSTRAINT fk_secrets_owner_to_groups_id FOREIGN KEY (owner) REFERENCES groups (id);
```
Comment (mmontgomery-square) [inline server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql:2]: We don't typically define foreign key constraints in our migration scripts, but it feels like the right thing to do.

**Comment by mmontgomery-square** [comment]
Well that didn't work. Turns out that our database tests (which I obviously neglected to run) rely on issuing truncate statements between executions in order to reset things to a pristine state, and you can't truncate a table which has a foreign key constraint (even if that operation would not violate the constraint, apparently):

```
SQL [truncate table groups]; Cannot truncate a table referenced in a foreign key constraint (`keywhizdb_test`.`secrets`, CONSTRAINT `fk_secrets_owner_to_groups_id` FOREIGN KEY (`owner`) REFERENCES `keywhizdb_test`.`groups` (`id`))
```

File: server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE secrets ADD owner BIGINT;
```
[the remark this answers is not in the document]
Comment (mmontgomery-square) [reply server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql:1]: It's a reference to a Group, and their primary key is a bigint. In the original version I had also added a foreign key constraint to make this clear, but unfortunately it's not compatible with the truncation we do during unit tests so I had to remove it.

[the remark this answers is not in the document]
Comment (isemaya-square) [reply server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql:1]: Ah ok, got it. Thanks for clarifying!

**Review by isemaya-square — APPROVED** [review]

File: server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (isemaya-square) [reply server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql:1]: (is there a way to put in a comment saying that it should be a foreign key but isn't compatible)?

[the remark this answers is not in the document]
Comment (mmontgomery-square) [reply server/src/main/resources/db/mysql/migration/V14__secrets_owner.sql:1]: Yeah, that's a good point - the structure of the relationships isn't really apparent from the tables, everything is disconnected. I wanted to add the foreign key not just as documentation but to ensure that you could only use a value that actually referred to an existing group. Honestly, I'd really like to fix up our tests such that this isn't a concern, but I haven't spent time on determining what the best path forward is.

**Pull request merged**